storage/rocksdb/ha_rocksdb.cc

```diff
@@ -10850,6 +11331,15 @@ int ha_rocksdb::update_write_sk(const TABLE *const table_arg,
     old_key_slice = rocksdb::Slice(
         reinterpret_cast<const char *>(m_sk_packed_tuple_old), old_packed_size);
 
+    /* Range locking: lock the index tuple being deleted */
+    if (m_use_range_locking) {
```

Comment:
It seems like this lock should go into check_and_lock_sk() instead?

Reply:
Also, it seems like this check should be using  `rocksdb_use_range_locking` instead? If the range lock manager is being used, then a singlepoint lock is needed on each secondary key regardless of the current transaction's isolation level.

For example:

```
TXN 1:

create table i1 (pk int, a int, b int, primary key (a), key(a)) engine=innodb;
insert into i1 values (1, 100, 1000), (2, 200, 2000), (3, 300, 3000);
set transaction isolation level repeatable read;
begin;
update i1 set b = b + 1 where a > 200;

TXN 2:

set transaction isolation level read committed;
begin;
insert into i1 values (4, 150, 1500); /* Success */
insert into i1 values (5, 250, 1500); /* Blocks */
```
The repeatable-read transaction would hold a range lock which potentially conflicts with

Reply:
> Also, it seems like this check should be using rocksdb_use_range_locking instead? 
Agree, done.